lte/gateway/python/integ_tests/s1aptests/test_attach_detach_with_non_nat_dhcp.py

```diff
@@ -0,0 +1,66 @@
+"""
+Copyright 2022 The Magma Authors.
+
+This source code is licensed under the BSD-style license found in the
+LICENSE file in the root directory of this source tree.
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+
+import unittest
+
+import s1ap_types
+from integ_tests.s1aptests import s1ap_wrapper
+from s1ap_utils import MagmadUtil
+
+
+class TestAttachDetachWithNonNatDhcp(unittest.TestCase):
+
+    def setUp(self):
+        """Initialize before test case execution"""
+        self.magma_utils = MagmadUtil(None)
+
+        self.magma_utils.enable_dhcp_config()
+        self.magma_utils.disable_nat()
+        self._s1ap_wrapper = s1ap_wrapper.TestWrapper()
+
+    def tearDown(self):
+        """Cleanup after test case execution"""
+        self._s1ap_wrapper.cleanup()
```

Comment:
I just had  the case where the test failed and didn't clan up the dhcp and nat config, not sure why it happened. Is it possible this cleanup method fails and then the steps after that are not run?

Reply:
But even if it does not fail, it might be better to adapt the mconfig first and then do the rest of the cleanup and restart all services.